sda-commons-client-jersey/src/test/java/org/sdase/commons/client/jersey/ClientProxyTest.java

```diff
@@ -10,50 +10,48 @@
 import static org.assertj.core.api.Assertions.assertThat;
 
 import com.github.tomakehurst.wiremock.http.RequestMethod;
-import com.github.tomakehurst.wiremock.junit.WireMockClassRule;
 import com.github.tomakehurst.wiremock.matching.RequestPatternBuilder;
-import io.dropwizard.testing.junit.DropwizardAppRule;
+import io.dropwizard.testing.junit5.DropwizardAppExtension;
 import javax.ws.rs.client.Client;
 import javax.ws.rs.core.Response;
 import org.apache.http.HttpHeaders;
 import org.apache.http.HttpStatus;
-import org.junit.Before;
-import org.junit.ClassRule;
-import org.junit.Test;
-import org.junit.rules.RuleChain;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
 import org.sdase.commons.client.jersey.test.ClientTestApp;
 import org.sdase.commons.client.jersey.test.ClientTestConfig;
-import org.sdase.commons.server.testing.SystemPropertyRule;
+import org.sdase.commons.client.jersey.wiremock.testing.WireMockClassExtension;
 
 /** A test that checks if the proxy can be configured via system properties */
-public class ClientProxyTest {
-  private static final WireMockClassRule CONTENT_WIRE =
-      new WireMockClassRule(wireMockConfig().dynamicPort());
-
-  private static final WireMockClassRule PROXY_WIRE =
-      new WireMockClassRule(wireMockConfig().dynamicPort());
-
-  private static final SystemPropertyRule PROP =
-      new SystemPropertyRule()
-          .setProperty("http.proxyHost", "localhost")
-          .setProperty("http.proxyPort", () -> "" + PROXY_WIRE.port())
-          .setProperty("http.nonProxyHosts", "localhost");
-
-  private static final DropwizardAppRule<ClientTestConfig> DW =
-      new DropwizardAppRule<>(ClientTestApp.class, resourceFilePath("test-config.yaml"));
-
-  @ClassRule
-  public static final RuleChain rule =
-      RuleChain.outerRule(CONTENT_WIRE).around(PROXY_WIRE).around(PROP).around(DW);
-
-  @Before
-  public void before() {
+class ClientProxyTest {
+  @RegisterExtension
+  @Order(0)
+  private static final WireMockClassExtension CONTENT_WIRE =
+      new WireMockClassExtension(wireMockConfig().dynamicPort());
+
+  @RegisterExtension
+  @Order(1)
+  private static final WireMockClassExtension PROXY_WIRE =
+      new WireMockClassExtension(wireMockConfig().dynamicPort());
+
+  @RegisterExtension
+  @Order(2)
+  static final DropwizardAppExtension<ClientTestConfig> DW =
+      new DropwizardAppExtension<>(ClientTestApp.class, resourceFilePath("test-config.yaml"));
+
+  @BeforeEach
+  void before() {
     CONTENT_WIRE.resetAll();
     PROXY_WIRE.resetAll();
+    System.setProperty("http.proxyHost", "localhost");
+    System.setProperty("http.proxyPort", "" + PROXY_WIRE.port());
+    System.setProperty("http.nonProxyHosts", "localhost");
```

Comment:
Rather use the [system property extension](https://junit-pioneer.org/docs/system-properties/).